### PR TITLE
Fix CLI directory separation and bash tool spec configuration

### PR DESCRIPTION
**Pull request by malhotra5**
- [ ] This change is worth documenting at https://docs.all-hands.dev/
- [x] Include this change in the Release Notes. If checked, you **must** provide an **end-user friendly** description for your change below

**End-user friendly description of the problem this fixes or functionality this introduces.**

This fix resolves issues with the OpenHands CLI where:
1. Configuration files and working directories were mixed together, causing confusion
2. The bash tool would use incorrect working directories when loaded from saved agent configurations

After this fix:
- Configuration files are properly stored in `~/.openhands` 
- Agent operations use the current working directory where the CLI is executed
- The bash tool always operates in the correct directory context
- Users can run the CLI from any directory and it will work correctly

---
**Summarize what the PR does, explaining any non-trivial design decisions.**

This PR implements two key fixes for the OpenHands CLI:

## 1. Directory Separation
**Problem**: Both `WORK_DIR` and `CONFIGURATIONS_DIR` pointed to the same location (`~/.openhands`), causing operational and configuration concerns to be mixed.

**Solution**: 
- `CONFIGURATIONS_DIR`: `~/.openhands` - Used for storing agent settings and CLI configuration files
- `WORK_DIR`: `os.getcwd()` - The current working directory where the CLI is executed, used for agent operations

**Files Modified**:
- `openhands_cli/locations.py`: Split the directory definitions
- `openhands_cli/tui/settings/settings_screen.py`: Updated to use appropriate directories
- `openhands_cli/tui/settings/store.py`: Updated AgentStore to use config directory

## 2. Dynamic Bash Tool Working Directory
**Problem**: When loading agent specifications from storage, the BashTool would retain whatever `working_dir` was saved in the spec, which could be incorrect for the current execution context.

**Solution**: Modified `AgentStore.load()` to dynamically update the BashTool's `working_dir` parameter to use the current working directory when loading agent specifications.

**Design Decision**: This ensures that regardless of where an agent configuration was created, the bash tool will always operate in the directory where the CLI is currently being executed.

## Testing
Added comprehensive test suite (`tests/test_directory_separation.py`) with 10 test cases covering:
- Directory separation verification
- BashTool spec updating behavior
- Edge cases (empty tools, missing params)
- Integration testing

All existing tests continue to pass (63 total tests passing).

---
**Link of any specific issues this addresses:**

This addresses the requirements specified in the user request to ensure `WORK_DIR` and `CONFIGURATIONS_DIR` are separate, and that the bash tool spec uses the current directory instead of hardcoded paths.

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:89ea249-nikolaik   --name openhands-app-89ea249   docker.all-hands.dev/all-hands-ai/openhands:89ea249
```

CLI with uvx:
```
uvx --python 3.12 --from git+https://github.com/All-Hands-AI/OpenHands@fix/cli-directory-separation-and-bash-tool-spec openhands
```

### CODE REVIEW

**Comment by malhotra5** [comment]
@openhands 

1. remove openhands/code from the pythonpath
2. read the diff of this PR
3. make sure that in addition to `BashTool` we're also updating the working dir for `FileEditorTool`
4. make sure this is checked for in the unit tests as well
5. we've rename `CONFIGURATIONS_DIR` to `PERSISTENCE_DIR`, so if appropriate rename method names or comments to reflect this new name as well

File: openhands-cli/openhands_cli/tui/settings/store.py [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+                    updated_tools.append(updated_tool_spec)
+                elif tool_spec.name == "FileEditorTool":
+                    # Update the workspace_root parameter to use current directory
+                    updated_params = tool_spec.params or {}
+                    updated_params["workspace_root"] = WORK_DIR
+                    updated_tool_spec = tool_spec.model_copy(update={"params": updated_params})
+                    updated_tools.append(updated_tool_spec)
+                else:
+                    updated_tools.append(tool_spec)
+            agent = agent.model_copy(update={"tools": updated_tools})
```
Comment (xingyaoww) [inline openhands-cli/openhands_cli/tui/settings/store.py]: oh man this is sad :(

Comment (xingyaoww) [reply openhands-cli/openhands_cli/tui/settings/store.py]: actually, can we reconstruct `tools` directly by calling create_default_tools and pass in work_dir, and use the result of that to override tools? this should save us some duplication

Comment (malhotra5) [reply openhands-cli/openhands_cli/tui/settings/store.py]: ah right good call

@openhands 

1. remove openhands/code from the pythonpath
2. look into the agent-sdk dependency for the the `create_default_tools` function
3. update this so that we're overriding the entire tool_spec field with the new tools returned from `create_default_tools` 
4. update the test in this PR (read the diff) to reflect this new logic
5. make sure tests are passing
